src/app/dynamic-dashboard/dynamic-dashboard.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, OnInit } from '@angular/core';
+import { ComponentService } from '../services/component.service';
+import { DSMService } from '../services/dsm.service';
+
+@Component({
+  selector: 'app-dynamic-dashboard',
+  templateUrl: './dynamic-dashboard.component.html',
+  styleUrls: ['./dynamic-dashboard.component.css']
+})
+export class DynamicDashboardComponent implements OnInit {
+
+  static HEATMAP_ROWS_LENGTH = 25;
+
+  enrolledParticipants: number;
+  registeredParticipants: number;
+  statistics: Array<any>;
+  
+  showParticipantInformation: boolean = false;
+
+  //fusion charts
+  title: String;
+  dataSource: Object;
+  dataSet = [];
+  rowsDictionary = {};
+  rows = [];
+  columns = {};
+  colorRange = {};
+  dataLoaded: boolean;
+  defaultRowLength = DynamicDashboardComponent.HEATMAP_ROWS_LENGTH;
+  numberOfParticipants: number;
+
+  constructor(private dsmService: DSMService) {
+    
+  }
+
+  ngOnInit() {
+    this.dsmService.getNumberOfParticipants(localStorage.getItem( ComponentService.MENU_SELECTED_REALM )).subscribe(
+      data => {
+        this.numberOfParticipants = data;
+      },
+      err => {
+
+      }
+    );
+    this.fetchStatistic(0, DynamicDashboardComponent.HEATMAP_ROWS_LENGTH);
```

Comment:
should that move up after L39 in case fetchStatistic returns result faster than getNumberOfParticipants?